tools/downloader: align namespace

jkamelin opened the pull request:
This PR is to align OMZ tools namespace in OpenVINO Python environment. The tools will now be located under openvino.model_zoo namespace and accessible through command line as omz_downloader, omz_converter, omz_quantizer, omz_data_downloader and omz_info_dumper commands, just like it is in openvino-dev PyPi package starting from OpenVINO 2021.4 release

vladimir-dudnik commented:
@nignatovsky we probably need to make change in jenkins job to replace direct call to downloader.py with install it and call to omz_downloader ?

@jkamelin what do you think?

jkamelin commented:
> @nignatovsky we probably need to make change in jenkins job to replace direct call to downloader.py with install it and call to omz_downloader ?
> 
> @jkamelin what do you think?

I agree

nignatovsky commented:
> @nignatovsky we probably need to make change in jenkins job to replace direct call to downloader.py with install it and call to omz_downloader ?
> 
> @jkamelin what do you think?

Yes, I will make necessary changes in the Downloader job

helena-intel commented on tools/downloader/README.md:
````diff
@@ -29,11 +29,17 @@ future releases.
+
+```sh
+python3 setup.py build
+python3 setup.py install
````
These instructions are for Ubuntu only. On Windows, if you install Python with the installer from python.org (the recommended way to install Python on Windows), setuptools and pip are included (though they are not always at the latest version). But if you type `python3` in a command prompt, you get redirected to the Windows Store to install Python. So on Windows you have to type `python`, but on Ubuntu, you have to type `python3`. 

This is how the PyPI openvino-dev install docs do this: https://github.com/openvinotoolkit/openvino/blob/master/docs/install_guides/pypi-openvino-dev.md 
For the notebooks we recommend a virtualenv and have different headings https://github.com/openvinotoolkit/openvino_notebooks/

jkamelin replied:
Changed command to `python`, added note about the difference on Linux

slyubimt commented on tools/model_tools/README.md:
````diff
@@ -27,15 +27,23 @@ future releases.
 
 ```sh
-python3 -mpip install --user -r ./requirements.in
+python setup.py build
````
Do we really need `python setup.py build`? `install` options covers this step.

jkamelin replied:
`install` option doesn't cover coping [open_model_zoo/models ](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models)and [open_model_zoo/data](https://github.com/openvinotoolkit/open_model_zoo/tree/master/data) folders to the package, since they are outside the project directory. These are needed for downloading and converting models.

slyubimt replied:
Consider to overload 'build_py' class instead of 'build':

```
26  import distutils.command.build    -> import distutils.command.build_py

57 class CustomBuild(distutils.command.build.build):    -> class CustomBuild(distutils.command.build_py.build_py):  

103   cmdclass={                         ->          cmdclass={
           'build': CustomBuild,                             'build_py': CustomBuild,
```

jkamelin replied:
Thanks, done

slyubimt commented on tools/model_tools/setup.py:
```diff
@@ -58,7 +58,7 @@ class CustomBuild(distutils.command.build.build):
         super().run()
 
-        package_build_dir = Path(self.build_lib, 'open_model_zoo/model_tools')
+        package_build_dir = Path(self.build_lib, 'openvino/model_zoo')
```
Please add __init__.py under ` openvino` folder with content like shown in a link below to support [pkgutil-style namespace packages](https://packaging.python.org/guides/packaging-namespace-packages/#pkgutil-style-namespace-packages)

https://github.com/openvinotoolkit/openvino/blob/master/tools/benchmark_tool/openvino/__init__.py

jkamelin replied:
Done

jkamelin commented:
> It is very welcome that the names are now aligned between pip and other installers! The README still seems to assume that a user needs to install open model zoo. Could it be clarified that that step is not necessary if a user already installed OpenVINO with pip, or OMZ tools with IRC? I find it useful to refer to the OMZ tools README for usage info about the tools (and people may arrive their when searching for how to use, say, `omz_quantizer`, but that installation step may lead people to think that they still have to do these steps too.

Added note that OMZ tools can be installed as a part of OpenVINO and provided installation steps are used to install from source. 

> Also, not essential but IMO it would be nicer if the README was not under `/tools/downloader/README`. It suggests that it is mostly about downloader, instead of all the OMZ tools. Is it being considered to change that, now that these relatively big changes are being made?

Renamed folder `/tools/downloader/` to `/tools/model_tools/`

jkamelin commented on tools/model_tools/setup.cfg:
```diff
@@ -30,8 +30,8 @@ where = src
-    omz_info_dumper = open_model_zoo.model_tools.info_dumper:main
-    omz_quantizer = open_model_zoo.model_tools.quantizer:main
+    omz_converter = openvino.model_zoo.omz_converter:main
+    omz_data_downloader = openvino.model_zoo.omz_data_downloader:main
```
Yes

vladimir-dudnik commented:
@jkamelin precommit check failed
12:42:28  tools/accuracy_checker/accuracy_checker/annotation_converters/README.md: unknown/disallowed HTML element in '<name>'
12:42:28  tools/accuracy_checker/accuracy_checker/annotation_converters/README.md: unknown/disallowed HTML element in '<index>'

vladimir-dudnik commented:
@nignatovsky note, this PR also affect Demo validation job

helena-intel commented:
What happens if a user installed OMZ according to the installation instructions and then a few months later installs openvino-dev, with a newer release? Is there a risk that their `omz_downloader` etc. then still points to a local directory (because that is earlier in the PATH) @slyubimt ? And is there a risk that things break in that case?

slyubimt commented:
>What happens if a user installed OMZ according to the installation instructions and then a few months later installs openvino-dev, with a newer release? Is there a risk that their omz_downloader etc. then still points to a local directory (because that is earlier in the PATH) @slyubimt ? And is there a risk that things break in that case?
If we consider the options to install both the packages, this one and openvino-dev, last installed overwrites aliases like  omz_downloader.

vladimir-dudnik commented:
Jenkins please retry a build

ilya-lavrenov commented:
BTW, why scripts have name `omz_downloader.py` etc while they already in `openvino.model_zoo`? See https://github.com/jkamelin/open_model_zoo/tree/jk/namespace/tools/model_tools/src/openvino/model_zoo

vladimir-dudnik commented:
@ilya-lavrenov names omz_downloader, omz_converter maintained for consistency with openvino-dev PyPi package in 2021.4. They will be called from command line in openvino-dev environment as omz_downloader. Names like simple downloader will look to generic in this case.

@jkamelin please rebase your branch

vladimir-dudnik commented:
14:16:05  models/public/yolox-tiny/README.md: URL reference "../../../tools/downloader/README.md" target does not exist or is not a file

vladimir-dudnik commented:
Jenkins please retry a build
